### DISCUSSION TITLE
Backporting attributes vulnerability fix to v.6.x (CVE-2023-22578)

### DISCUSSION THREAD

**Discussion by panusoi** [sequelize/sequelize#15694] [Help & Questions] [answered]
Hi, our security scan reported a vulnerability in Sequelize v.6.28.0 (CVE-2023-22578). I think this is related to this pr: https://github.com/sequelize/sequelize/pull/15374, I tested with v.6.28.0 and confirmed that the vulnerability exists.

The question is that are you going to backport the fix to v.6.x?

**Comment by ephys** [comment, marked as answer]
Hi!

At the present time I do not know.

The problem is that, while I did make the changes to remove a footgun in v7, that behavior was not a bug but a deliberate feature implemented by one of the previous teams that worked on Sequelize, and there are codebases relying on it.

Backporting #15374 would introduce a breaking change in a minor release. We've exceptionally done that in the past when we fixed a major issue (#14519), but I am not convinced that it's warranted for this one.

The reason I think releasing this breaking change in a minor release is unwarranted is that it can only be a problem if you use user-provided values as the name of a column to select. Even without the fix, you should make sure that input is safe as the query will crash if the input is not a valid attribute name.

We're stuck between a rock and a hard place. We can either introduce a breaking change in a minor release, or have a critically vulnerability warning.

We need to check with the people that published that vulnerability if they would consider a big bold warning that that property accepts arbitrary SQL (and therefore that user data should not be used in it) to be sufficient to dismiss the audit warning

Link to github advisory: https://github.com/advisories/GHSA-8mwq-mj73-qv68

**Reply by usmansagri** [reply]
Hi @ephys ,

Good day!

We are also facing same vulnerability in v6.6.2 and on upgrading to v6.29.0 we're getting errors in files like model.d.ts, bulk-record-error.d.ts, validation-error.d.ts, sequelize.d.ts

I have also dropped you an email personally containing screenshots from sagri.usman@gmail.com to zoe@ephys.dev

Kindly suggest how to fix these errors.
![sequelize](https://user-images.githubusercontent.com/32294868/222065623-507cdcdf-104e-4707-be9f-6fd228862393.png)

**Reply by ephys** [reply]
I'm hiding your comment as it's already handled in [this thread](https://github.com/sequelize/sequelize/discussions/15723) to not distract from the purpose of this thread

**Comment by ephys** [comment]
I've opened a PR to fix this issue: https://github.com/sequelize/sequelize/pull/15710. We opted for what we believe to be the lesser of the two evils

**Comment by ephys** [comment]
We have now released that change as Sequelize 6.29.0
